### ISSUE TITLE
Swagger file is invalid

### ISSUE THREAD

**Issue by Gontran08** [jimschubert/intellij-openapi-generator#19]
When I try to generate code from my swagger file, I always have an error saying that my file is not valid. It works when I use the Swagger Editor.

![capture d ecran de 2018-07-25 12-01-04](https://user-images.githubusercontent.com/7196461/43212654-8a246704-9002-11e8-84b4-b6f0f1833363.png)

Here is my file:
```
openapi: "3.0.0"
info:
  version: 0.0.1
  title: LINK API
servers:
  - url: http://link.koremtest.com/v1
paths:
  /map/layers:
    get:
      summary: Get all layers
      operationId: listLayers
      tags:
        - layers
      responses:
        '200':
          description: A list of layers
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/LayerDefinition"
                minItems: 1
        default:
          description: Request status
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Status"
    post:
      summary: Create a layer
      operationId: createLayer
      tags:
        - layers
      requestBody:
        description: The layer to create
        required: true
        content:
          application/json:
            schema:
              oneOf:
                - $ref: '#/components/schemas/NativeLayerDefinition'
                - $ref: '#/components/schemas/DynamicLayerDefinition'
      responses:
        '200':
          description: New layer definition
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/LayerDefinition'
        default:
          description: Request status
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Status"
  /map/layers/{layerId}:
    get:
      summary: Get a specific layer
      operationId: getLayerById
      tags:
        - layers
      parameters:
        - name: layerId
          in: path
          required: true
          description: The id of the layer to retrieve
          schema:
            type: string
      responses:
        '200':
          description: Layer with the specified id
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/LayerDefinition"
        default:
          description: Request status
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Status"
    patch:
      summary: Update a specific layer
      operationId: updateLayerById
      tags:
        - layers
      parameters:
        - name: layerId
          in: path
          required: true
          description: The id of the layer to update
          schema:
            type: string
      requestBody:
        description: The new layer definition
        required: true
        content:
          application/json:
            schema:
              oneOf:
                - $ref: '#/components/schemas/NativeLayerDefinition'
                - $ref: '#/components/schemas/DynamicLayerDefinition'
      responses:
        '200':
          description: New layer definition
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/LayerDefinition"
        default:
          description: Request status
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Status"
    delete:
      summary: Delete a specific layer
      operationId: deleteLayerById
      tags:
        - layers
      parameters:
        - name: layerId
          in: path
          required: true
          description: The id of the layer to delete
          schema:
            type: string
      responses:
        default:
          description: Request status
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Status"
  /map/layers/delete:
    post:
      summary: Delete a list of layers
      operationId: deleteLayers
      tags:
        - layers
      requestBody:
        description: Layer to create
        required: true
        content:
          application/json:
            schema:
              type: array
              items:
                type: string
              minItems: 1
      responses:
        default:
          description: Request status
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Status"
  /map/layers/update:
    post:
      summary: Update a list of layers
      operationId: updateLayers
      tags:
        - layers
      requestBody:
        description: Layers to update
        required: true
        content:
          application/json:
            schema:
              type: array
              items:
                anyOf:
                  - $ref: '#/components/schemas/NativeLayerDefinition'
                  - $ref: '#/components/schemas/DynamicLayerDefinition'
              minItems: 1
      responses:
        default:
          description: Request status
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Status"
components:
  schemas:
    LayerDefinition:
      required:
        - name
      properties:
        id:
          type: string
        name:
          type: string
        metadata:
          readOnly: true
          allOf:
            - $ref: "#/components/schemas/FeatureMetadata"
        visible:
          type: boolean
    Status:
      required:
        - success
      properties:
        success:
          type: boolean
    KeyDefinition:
      required:
        - type
        - attributes
      properties:
        type:
          type: string
        attributes:
          type: array
          items:
            type: string
          minItems: 1
    TableMetadata:
      required:
        - supportsInsert
        - supportsDelete
        - supportsUpdate
        - KeyDefinition
      properties:
        supportsInsert:
          type: boolean
        supportsDelete:
          type: boolean
        supportsUpdate:
          type: boolean
        KeyDefinition:
          $ref: "#/components/schemas/KeyDefinition"
    Metadata:
      required:
        - name
        - type
      properties:
        name:
          type: string
        type:
          type: string
        bbox:
          type: array
          items:
            type: number
          minItems: 4
          maxItems: 4
        styleColumn:
          type: string
        fractionalDigits:
          type: integer
        totalDigits:
          type: integer
    FeatureMetadata:
      required:
        - Metadata
        - TableMetadata
      properties:
        Metadata:
          $ref: "#/components/schemas/Metadata"
        TableMetadata:
          $ref: "#/components/schemas/TableMetadata"
    SpatialAttribute:
      properties:
        srsName:
          type: string
    GeometryAttribute:
      allOf:
        - $ref: "#/components/schemas/SpatialAttribute"
        - type: object
          required:
            - geometryColumn
          properties:
            geometryColumn:
              type: string
            typeColumn:
              type: string
            wkbColumn:
              type: string
            xminColumn:
              type: string
            yminColumn:
              type: string
            xmaxColumn:
              type: string
            ymaxColumn:
              type: string
            predominantGeometryType:
              type: string
    XYAttribute:
      allOf:
        - $ref: "#/components/schemas/SpatialAttribute"
        - type: object
          required:
            - longitudeColumn
            - latitudeColumn
          properties:
            longitudeColumn:
              type: string
            latitudeColumn:
              type: string
    NativeLayerDefinition:
      allOf:
        - $ref: "#/components/schemas/LayerDefinition"
        - type: object
          required:
            - fileName
          properties:
            fileName:
              type: string
    StyleAttribute:
      required:
        - styleColumn
        - type
      properties:
        styleColumn:
          type: string
        type:
          type: string
    DynamicLayerDefinition:
      allOf:
        - $ref: "#/components/schemas/LayerDefinition"
        - type: object
          required:
            - primaryKey
            - table
            - query
            - pool
            - spatialAttributes
          properties:
            primaryKey:
              type: string
            table:
              type: string
            query:
              type: string
            pool:
              type: string
            sqlOptions:
              type: string
            searchType:
              type: string
            dpi:
              type: integer
            spatialAttributes:
              type: array
              items:
                oneOf:
                  - $ref: "#/components/schemas/GeometryAttribute"
                  - $ref: "#/components/schemas/XYAttribute"
            styleAttribute:
              $ref: "#/components/schemas/StyleAttribute"
```

**Comment by jimschubert** [comment]
@Gontran08 Sorry I didn't see your issue until just now, answering a newer one.

This plugin didn't support version 3.0.0 specs until version 2.0.0 of the plugin. Unfortunately, I messed up the publish and had set the supported IDE version to 2017.3.x only. I've just pushed a new version (2.1.0) which updates to OpenAPI Generator 3.2.0 and supports IDE versions 2016.2+. It'll take a day or two for Jetbrains to approve.

For reference, the other issue (#20) reported the IDE version problem.
